EIPS/eip-5007.md

```diff
@@ -0,0 +1,313 @@
+
+There are also some NFTs contain time functions, but the naming is different, third-party platforms are difficult to develop based on it.
+
+By introducing (`startTime`, `endTime`) and unifying the naming, it is possible to enable and disable NFT automatically on chain.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+interface IERC5007  {
+
+    /// @notice Emitted when the `startTime` or `endTime` of a NFT is changed
+    /// @param tokenId  The tokenId of the NFT
+    /// @param startTime  The new start time of the NFT
+    /// @param endTime  The new end time of the NFT
+    event TimeUpdate(uint256 tokenId, uint64 startTime, uint64 endTime);
+
+    /// @notice Get the start time of the NFT
+    /// @dev Throws if `tokenId` is not valid NFT
+    /// @param tokenId  The tokenId of the NFT
+    /// @return The start time of the NFT
+    function startTime(uint256 tokenId) external view returns (uint64);
+    
+    /// @notice Get the end time of the NFT
+    /// @dev Throws if `tokenId` is not valid NFT
+    /// @param tokenId  The tokenId of the NFT
+    /// @return The end time of the NFT
+    function endTime(uint256 tokenId) external view returns (uint64);
+
+    /// @notice Get the token id which this NFT mint from
+    /// @dev Throws if `tokenId` is not valid NFT
+    /// @param tokenId  The tokenId of the NFT
+    /// @return The token id which this NFT mint from
+    function originalTokenId(uint256 tokenId) external view returns (uint256);
+
+    /// @notice Check the NFT is valid now 
+    /// @dev Throws if `tokenId` is not valid NFT
+    /// @param tokenId  The tokenId of the NFT
+    /// @return The the NFT is valid now
+    /// if(startTime <= now <= endTime) {return true;} else {return false;}
+    function isValidNow(uint256 tokenId) external view returns (bool);
+
+    /// @notice Mint a new token from an old token
+    /// @dev Throws if `tokenId` is not valid token
+    /// @param originalTokenId_  The token id which the new token mint from
+    /// @param newTokenStartTime  The start time of the new token
+    /// @param newTokenOwner  The owner of the new token
+    /// @return newTokenId The the token id of the new token
+    function split(uint256 originalTokenId_, uint64 newTokenStartTime, address newTokenOwner) external returns(uint256);
+
+    /// @notice Merge two time NFTs into one time NFT
+    /// @dev Throws if `firstTokenId` or `secondTokenId` is not valid token
+    /// @param firstTokenId   The id of the first token
+    /// @param secondTokenId  The id of the second token
+    /// @param newTokenOwner  The owner of the new token
+    /// @return newTokenId The id of the new token
+    function merge(uint256 firstTokenId, uint256 secondTokenId, address newTokenOwner) external returns(uint256);
+}
+```
+
```

Comment:
You'll want to include the EIP-165 interface id, and require that implementers implement `supportsInterface`